src/tracker.jl

```diff
@@ -0,0 +1,329 @@
+            $f(data(x1), data(x2)), (Δ) -> (Δ[1:length(x1)], Δ[length(x1)+1])
+        end
+    end
+end
+
+function Base.copy(
+    A::TrackedArray{T, 2, <:Adjoint{T, <:AbstractTriangular{T, <:AbstractMatrix{T}}}},
+) where {T <: Real}
+    return track(copy, A)
+end
+@grad function Base.copy(
+    A::TrackedArray{T, 2, <:Adjoint{T, <:AbstractTriangular{T, <:AbstractMatrix{T}}}},
+) where {T <: Real}
+    return copy(data(A)), ∇ -> (copy(∇),)
+end
+
+Base.:*(A::TrackedMatrix, B::AbstractTriangular) = track(*, A, B)
+Base.:*(A::AbstractTriangular{T}, B::TrackedVector) where {T} = track(*, A, B)
+Base.:*(A::AbstractTriangular{T}, B::TrackedMatrix) where {T} = track(*, A, B)
+Base.:*(A::Adjoint{T, <:AbstractTriangular{T}}, B::TrackedMatrix) where {T} = track(*, A, B)
+Base.:*(A::Adjoint{T, <:AbstractTriangular{T}}, B::TrackedVector) where {T} = track(*, A, B)
+
+function Base.fill(
+    value::TrackedReal,
+    dims::Vararg{Union{Integer, AbstractUnitRange}},
+)
+    return track(fill, value, dims...)
+end
+@grad function Base.fill(value::Real, dims...)
+    return fill(data(value), dims...), function(Δ)
+        size(Δ) ≢  dims && error("Dimension mismatch")
+        return (sum(Δ), map(_->nothing, dims)...)
+    end
+end
+
+
+## StatsFuns ##
+
+logsumexp(x::TrackedArray; dims=:) = _logsumexp(x, dims)
+_logsumexp(x::TrackedArray, dims=:) = track(_logsumexp, x, dims)
+@grad function _logsumexp(x::TrackedArray, dims)
+    lse = logsumexp(data(x), dims = dims)
+    return lse, Δ -> (Δ .* exp.(x .- lse), nothing)
+end
+
+
+## Linear algebra ##
+
+# Work around https://github.com/FluxML/Tracker.jl/pull/9#issuecomment-480051767
+upper(A::AbstractMatrix) = UpperTriangular(A)
+lower(A::AbstractMatrix) = LowerTriangular(A)
+function upper(C::Cholesky)
+    if C.uplo == 'U'
+        return upper(C.factors)
+    else
+        return copy(lower(C.factors)')
+    end
+end
+function lower(C::Cholesky)
+    if C.uplo == 'U'
+        return copy(upper(C.factors)')
+    else
+        return lower(C.factors)
+    end
+end
+
+LinearAlgebra.LowerTriangular(A::TrackedMatrix) = lower(A)
+lower(A::TrackedMatrix) = track(lower, A)
+@grad lower(A) = lower(Tracker.data(A)), ∇ -> (lower(∇),)
+
+LinearAlgebra.UpperTriangular(A::TrackedMatrix) = upper(A)
+upper(A::TrackedMatrix) = track(upper, A)
+@grad upper(A) = upper(Tracker.data(A)), ∇ -> (upper(∇),)
+
+function LinearAlgebra.cholesky(A::TrackedMatrix; check=true)
+    factors_info = turing_chol(A, check)
+    factors = factors_info[1]
+    info = data(factors_info[2])
+    return Cholesky{eltype(factors), typeof(factors)}(factors, 'U', info)
+end
+function LinearAlgebra.cholesky(A::Symmetric{<:Any, <:TrackedMatrix}; check=true)
+    uplo = A.uplo == 'U' ? (:U) : (:L)
+    factors_info = symm_turing_chol(parent(A), check, uplo)
+    factors = factors_info[1]
+    info = data(factors_info[2])
+    return Cholesky{eltype(factors), typeof(factors)}(factors, 'U', info)
+end
```

Comment:
Good to keep in mind that these lines (and many more in this file and in other files for other AD backends) should be moved upstream and are type piracy.